jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -544,14 +544,16 @@ public void handle()
                         // RFC 7230, section 3.3.  We do this here so that a servlet error page can be sent.
                         if (!_servletContextRequest.isHead() && getServletContextResponse().getStatus() != HttpStatus.NOT_MODIFIED_304)
                         {
-                            long written = getBytesWritten();
-                            if (getServletContextResponse().isContentIncomplete(written) && sendErrorOrAbort("Insufficient content written %d < %d".formatted(written, getServletContextResponse().getContentLength())))
+                            long written = getHttpOutput().getWritten();
```

Comment:
We need a comment here saying why `getByteWritten()` is not used. i.e. because we want to look at the bytes written to the `HttpOutputStream` and not the bytes written to the underlying core response.

It would also be good to comment in `ServletContextResponse` that its `_contentLength` field is the length set by the application, but not necessarily the one set on the response as it may be removed/adjsuted.